src/BenchmarkDotNet.Core/Engines/IHostApi.cs

```diff
@@ -0,0 +1,20 @@
+﻿using System;
+
+namespace BenchmarkDotNet.Engines
+{
+    public interface IHostApi
```

Comment:
Are you sure that we need all these methods in the interface? For example, it's easy to implement `WriteLine` and his friends as extension methods via `Write`. The signal methods also could be moved out from the interface. If we are going to create other implementation of `IHostApi` (e.g. via sockets), such extension method will help to avoid code duplications.